src/indra_cogex/client/queries.py

```diff
@@ -1140,6 +1144,274 @@ def get_stmts_for_stmt_hashes(
+    agent_name : Optional[str], default: None
+        The name of the agent to filter by (e.g., "EGFR").
+    agent_role : Optional[str], default: None
+        The role of the agent in the interaction, either "subject" or "object".
+    limit : Optional[int], default: 10
+        The maximum number of statements returned
+    client : Neo4jClient
+        The Neo4j client to use for querying.
+    evidence_limit : Optional[int], default: None
+        The optional limit for the number of evidence entries per statement.
+
+    Returns
+    -------
+    List[Statement]
+        A list of statements filtered by the provided constraints.
+    """
+
+    if agent_name and agent_role:
+        if agent_role.lower() == "subject":
+            match_clause = f"(a:BioEntity {{name: $agent_name}})-[r:indra_rel"
+            match_direction = "]->(b:BioEntity)"
+        elif agent_role.lower() == "object":
+            match_clause = f"(a:BioEntity)-[r:indra_rel"
+            match_direction = f"]->(b:BioEntity {{name: $agent_name}})"
+    elif agent_name:
+        match_clause = f"(a:BioEntity {{name: $agent_name}})-[r:indra_rel"
+        match_direction = "]-(b:BioEntity)"
+    else:
+        match_clause = f"(a:BioEntity)-[r:indra_rel"
+        match_direction = "]-(b:BioEntity)"
+
+    match_clause += match_direction
+    where_clauses = []
+    if rel_types:
+        if isinstance(rel_types, str):
+            rel_types = [rel_types]
+        where_clauses.append("r.stmt_type IN $rel_types")
+    if stmt_sources:
+        if isinstance(stmt_sources, str):
+            stmt_sources = [stmt_sources]
+        source_conditions = " OR ".join(
+            [f'r.source_counts CONTAINS \'"{source}":\'' for source in
+             stmt_sources])
+        where_clauses.append(f"({source_conditions})")
+
+    if where_clauses:
+        match_clause += " WHERE " + " AND ".join(where_clauses)
+
+    query = f"MATCH p = {match_clause} RETURN p LIMIT $limit"
+
+    params = {
+        "agent_name": agent_name,
+        "rel_types": rel_types if isinstance(rel_types, list) else [rel_types],
+        "limit": limit
+    }
+
+    logger.info(f"Running query with constraints: rel_type={rel_types}, "
+                f"source={stmt_sources}, agent_name={agent_name}, "
+                f"agent_role={agent_role}, limit={limit}")
+    rels = client.query_relations(query, **params)
+    stmts = indra_stmts_from_relations(rels, deduplicate=True)
+
+    if evidence_limit and evidence_limit > 1:
+        stmts = enrich_statements(
+            stmts,
+            client=client,
+            evidence_limit=evidence_limit,
+        )
+    if not return_evidence_counts:
+        return stmts
+    evidence_counts = {
+        stmt.get_hash(): rel.data["evidence_count"]
+        for rel, stmt in zip(rels, stmts)
+    }
+
+    return stmts, evidence_counts
+
+@autoclient()
+def get_stmts_for_agent_type(
+    agent_name: str,
+    agent_role: str,
```

Comment:
If it's allowed to not specify a direction, this parameter should be optional.
```suggestion
    agent_role: Optional[str] = None,
```